test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/ViewDataDictionaryTest.cs

```diff
@@ -50,33 +49,26 @@ public void ConstructorInitalizesMembers()
         }
 
         [Fact]
-        public void SetModelUsesPassedInModelMetadataProvider()
+        public void Constructor_UsesModelMetadataProvider()
         {
             // Arrange
-            var metadataProvider = new Mock<IModelMetadataProvider>();
+            var metadataProvider = new Mock<IModelMetadataProvider>(MockBehavior.Strict);
             metadataProvider
                 .Setup(m => m.GetMetadataForType(typeof(object)))
                 .Returns(new EmptyModelMetadataProvider().GetMetadataForType(typeof(object)))
                 .Verifiable();
-            metadataProvider
-                .Setup(m => m.GetMetadataForType(typeof(TestModel)))
-                .Returns(new EmptyModelMetadataProvider().GetMetadataForType(typeof(TestModel)))
-                .Verifiable();
             var modelState = new ModelStateDictionary();
-            var viewData = new TestViewDataDictionary(metadataProvider.Object, modelState);
-            var model = new TestModel();
 
             // Act
-            viewData.SetModelPublic(model);
+            var viewData = new TestViewDataDictionary(metadataProvider.Object, modelState);
 
             // Assert
             Assert.NotNull(viewData.ModelMetadata);
-            metadataProvider.Verify();
+            metadataProvider.Verify(m => m.GetMetadataForType(typeof(object)), Times.Once());
         }
 
-        // When SetModel is called, only GetMetadataForType from MetadataProvider is expected to be called.
         [Fact]
-        public void SetModelCallsGetMetadataForTypeExactlyOnce()
+        public void SetModel_DoesNotUsePassedInModelMetadataProvider()
```

Comment:
I don't think this is exactly what's being verified - this seems to imply that there's a second MMP.

It's just not doing anything...